### ISSUE TITLE
My password is invalid???

### ISSUE THREAD

**Issue by aerozoic** [MinetestForFun/server-minetestforfun-skyblock#160]
@Darcidride I can't get in the game. My password mysteriously doesn't work :(

I'm now suspicious that the file holding passwords is easily corruptible.
:large_orange_diamond:

**Comment by BetterToAutomateTheWorld** [comment]
Fixed with **/setpassword cessna151 <value>**

**Comment by ElementW** [comment]
@cessna151 The file holding passwords is `auth.txt` which always had corruption issues, including lately due to the `SadieXYZ` influx. I don't know how Minetest's core implements that file's handling, but from a statistical point of view I'd say it gets corrupted for every 100~150 changes (new player, password change, priv changes); which only means a single thing: that code is pure, untested trash (like most of MT).

**Comment by Lymkwi** [comment]
@cessna151 and I discussed about a feature which could potentially prevent the flood of modifications made to `auths.txt` from corrupting our data. It is a mod which would record the join time of players and hold the timestamps in a table. After X minutes, any entry from a player would be removed. If the player left, their entry would be removed as well after a short period of time (~30 seconds, for example).
The idea is to prevent new players from joining when too many entries are already in our array, and ask them to wait for the purge of enough entries to make place for theirs.
It would only really solve it for joining players (ignoring auth modifications like last connection time - part, password changes, etc), and would be slightly weakened by the fact that entries for gone players would be removed rather quickly (compared to an interval of about 10 minutes for entries of players who stayed).
It is really the only thing i can think of. Either we code that, or someone spends the time to work on the engine and apply that patch onto our engine, even if it's not merged into the main branch. I'd honestly do that if I had enough knowledge in how to interact with squlite databases in C/C++ (they use C macros AFAIK in the engine's code), knowledge that I have in python though.. And something tells me it's not a great idea to copy what is already present in the engine for rollbacks and map saving, but if that's the only long-lasting solution, we've got to do it...

EDIT: OMFG I think I have an idea of what the heck is really stressing the auths. Privileges. Constantly read, granted, rewritten, etc.
